crc_errors/CRC_Count_check.py

```diff
@@ -28,50 +25,43 @@
 class common_setup(aetest.CommonSetup):
     """ Common Setup section """
 
-    # CommonSetup have subsection.
-    # You can have 1 to as many subsection as wanted
-
     # Connect to each device in the testbed
     @aetest.subsection
     def connect(self, testbed):
-        genie_testbed = Genie.init(testbed)
+        genie_testbed = load(testbed)
         self.parent.parameters['testbed'] = genie_testbed
         device_list = []
         for device in genie_testbed.devices.values():
             log.info(banner(
-                "Connect to device '{d}'".format(d=device.name)))
+                f"Connect to device '{device.name}'"))
             try:
                 device.connect()
             except Exception as e:
-                self.failed("Failed to establish connection to '{}'".format(
-                    device.name))
+                self.failed(f"Failed to establish connection to '{device.name}'")
 
             device_list.append(device)
 
         # Pass list of devices the to testcases
         self.parent.parameters.update(dev=device_list)
+        print(self.parent.parameters)
 
 
 ###################################################################
 #                     TESTCASES SECTION                           #
 ###################################################################
 
-# Testcase name : vxlan_consistency_check
+# Testcase name : Check for CRC errors
 class CRC_count_check(aetest.Testcase):
```

Comment:
now on 2nd thought what could be REALLY cool is to:

1. in the above connect section, keep the device_list, but use it for only devices we could "connect to"
2. instead of having a single crc count check testcase here, do a testcase loop:

  - for each device defined in the testbed yaml, create a new testcase
  - for each interface collected in the learn, generate a test

pseudo code:

```python

class CommonSetup(aetest.CommonSetup):
    @aetest.subsecton
    def create_testcases(self):
        aetest.loop.mark(CRC_count_check, device=device_list)

class CRC_count_Check(aetest.Testcase):
    @aetest.setup
    def setup(self, device): 
        self.interface_info = device.learn('interface')
        # do magic and build the mega dict
        # then
        aetest.loop.mark(self.interface_check, intf=list_of_interfaces)
   
    @aetest.test
    def interface_check(self, intf):
        # pass fail based on interface

```

this would give you a much more detailed result - each interface becomes a test in the result page

maybe it's a bit too much... just a thought :P

Reply:
So I understand.  Only create device_list after we have successfully connected to devices and populated with those devices.